### PR TITLE
Allow setting current resource in tests

### LINKED ISSUE #138: Testing Authenticated Elements
### Question

I'm currently trying to set this up, but I'm having difficulty running any form of integration test on the schema itself.

I have a number of tests in a format similar to [that specified by the GraphQL-Ruby documentation](https://graphql-ruby.org/testing/integration_tests.html#testing-interface-level-behaviors), but with this gem in operation, all my tests now fail with the claim that the user is not signed in.

I'm executing queries from the schema, not the controller, but the schema and controller are now more tightly coupled since the schema now delegates to the controller for authentication. Is there a recommended means by which a user can be readily specified for a schema call? I've been digging through the gem callchain and I can't seem to figure out how this might be done. I've tried creating a context for the `Schema#execute` call with a stubbed controller, but that doesn't seem to do the trick.

### PR DESCRIPTION
**Pull request by 00dav00**
Allows setting `current_resource` on specs using `with graphql schema test` shared context

Closes #138

### CODE REVIEW

File: lib/graphql_devise/schema_plugin.rb [hunk 1]
```diff
@@ -40,18 +40,21 @@ def trace(event, trace_data)
...
-      context[:current_resource] = resource_names.find do |resource_name|
-        unless Devise.mappings.key?(resource_name)
-          raise(
-            GraphqlDevise::Error,
-            "Invalid resource_name `#{resource_name}` provided to `graphql_context`. Possible values are: #{Devise.mappings.keys}."
-          )
+      controller     = context[:controller]
+      resource_names = Array(context[:resource_name])
+
+      if context[:current_resource].blank?
```
Comment (mcelicalderon) [inline lib/graphql_devise/schema_plugin.rb:46]: Why not the shorter version?

```ruby
context[:current_resource] ||= resource_names.find do |resource_name|

**Review by mcelicalderon — APPROVED** [review]
Looks good, just a small suggestion and don't forget to remove commented code.

**Comment by keithmgould** [comment]
This is super cool -- I see that its closed now but is there a way for new users of the gem to know about how to properly test? Maybe a readme section could be added show the methods included in this PR, or at least a link from readme to the above?

**Comment by 00dav00** [comment]
Hi @keithmgould , I think that is a great suggestion!
Would you like to create a PR with that addition to the README?

**Comment by mcelicalderon** [comment]
I can also do that if no one has done it until I get some time. Important thing to note here is that this is a quick fix that works for all but one scenario, passing no `current_resource` for an authenticated query. This will still throw an error unless you stub the controller injected in the context. A more complete fix is still on the works as it might require a bigger refactor of how we set the current user in the context and also the removal of queries and mutations that return http redirects in favor of simpler confirmation and password reset flows. This way we might be able not to require the controller to be injected all together

**Pull request merged**